angular/projects/researchdatabox/form/src/app/form-state/effects/form-event-bus-adapter.effects.ts

```diff
@@ -0,0 +1,302 @@
+/**
+ * Form Event Bus Adapter Effects
+ * 
+ * Promotes qualifying ephemeral bus events into persistent NgRx actions.
+ * Per R15.20–R15.29, AC37–AC44
+ * 
+ * Promotion Criteria:
+ * (a) Event affects persistent global state
+ * (b) Event triggers a side-effect requiring store coordination
+ * (c) Event requires replay capability for debugging
+ */
+
+import { Injectable, inject, InjectionToken } from '@angular/core';
+import { Actions, createEffect } from '@ngrx/effects';
+import { Store } from '@ngrx/store';
+import { Observable, EMPTY } from 'rxjs';
+import { map, throttleTime, tap, filter, catchError } from 'rxjs/operators';
+import { FormComponentEventBus } from '../events/form-component-event-bus.service';
+import { FormComponentEventType, FormSaveSuccessEvent } from '../events/form-component-event.types';
+import * as FormActions from '../state/form.actions';
+import { LoggerService } from '@researchdatabox/portal-ng-common';
+
+/**
+ * Configuration for the adapter
+ */
+export interface FormEventBusAdapterConfig {
+  /** Enable diagnostic logging (R15.26) */
+  diagnosticsEnabled?: boolean;
+  /** Throttling window in ms to prevent duplicates (R15.22) */
+  throttleWindowMs?: number;
+  /** Disable all promotions (R15.27) */
+  disabled?: boolean;
+}
+
+/**
+ * Injection token for adapter configuration
+ */
+export const FORM_EVENT_BUS_ADAPTER_CONFIG = new InjectionToken<FormEventBusAdapterConfig>(
+  'FORM_EVENT_BUS_ADAPTER_CONFIG'
+);
+
+/**
+ * Default configuration
+ */
+const DEFAULT_CONFIG: Required<FormEventBusAdapterConfig> = {
+  diagnosticsEnabled: false,
+  throttleWindowMs: 250,
+  disabled: false
+};
+
+/**
+ * Promotion criterion codes for diagnostics (R15.26)
+ */
+enum PromotionCriterion {
+  AFFECTS_STATE = 'a',        // Affects persistent global state
+  TRIGGERS_SIDE_EFFECT = 'b', // Triggers a side-effect
+  REQUIRES_REPLAY = 'c'       // Requires replay in debugging
+}
+
+/**
+ * Logs promotion decisions in diagnostic mode (R15.26)
+ */
+function logPromotion(logger: LoggerService, eventType: string, criterion: PromotionCriterion, actionType: string): void {
+  logger.debug(
+    `[FormEventBusAdapter] Promoting event → action`,
+    { eventType, criterion, actionType }
+  );
+}
+
+/**
+ * Logs skipped events in diagnostic mode
+ */
+function logSkipped(logger: LoggerService, eventType: string, reason: string): void {
+  logger.debug(
+    `[FormEventBusAdapter] Skipped event`,
+    { eventType, reason }
+  );
+}
+
+/**
+ * Formats validation errors for readable messages in actions
+ * - If errors is an object: "key: message" pairs joined by ", "
+ * - If errors is an array: messages joined by "; "
+ * - If errors is a string: returned as-is
+ * - Otherwise: JSON stringified
+ */
+function formatErrorsForMessage(errors: any): string {
```

Comment:
This form validation errors to readable messages is ok for now. It will need to be updated along with the changes to the server-side validation errors.